torch/__init__.py

```diff
@@ -275,26 +275,50 @@ def _load_dll_libraries() -> None:
     del _load_dll_libraries
 
 
-def _preload_cuda_deps(lib_folder: str, lib_name: str) -> None:
-    """Preloads cuda deps if they could not be found otherwise."""
-    # Should only be called on Linux if default path resolution have failed
-    assert platform.system() == "Linux", "Should only be called on Linux"
-
-    lib_path = None
-    for path in sys.path:
-        nvidia_path = os.path.join(path, "nvidia")
-        if not os.path.exists(nvidia_path):
-            continue
-        candidate_lib_paths = glob.glob(
-            os.path.join(nvidia_path, lib_folder, "lib", lib_name)
-        )
-        if candidate_lib_paths and not lib_path:
-            lib_path = candidate_lib_paths[0]
-        if lib_path:
-            break
-    if not lib_path:
-        raise ValueError(f"{lib_name} not found in the system path {sys.path}")
-    ctypes.CDLL(lib_path)
+def _preload_pypi_cuda_deps() -> None:
+    """Try to preloads cuda deps if possible."""
+    # Only a Linux issue for now
+    if platform.system() != "Linux":
```

Comment:
Why not assert here?

Reply:
it's possible `platform.system() == "Darwin"` here

Reply:
The old logic here only search for site-package/nvidia libs when `ctypes.CDLL(global_deps_lib_path, mode=ctypes.RTLD_GLOBAL)` failed, and this is a linux-only case. Current implementation always tries to preload pypi nv libs.